File: src/CombatAnalysis.WebApp/ClientApp/src/services/FriendService.js
import HttpClient from '../helpers/HttpClient'

class FriendService {
    _httpClient = null;

    constructor() {
        this._httpClient = new HttpClient();
    }

    async getAllAsync() {
        try {
            const result = await this._httpClient.getAsync("Friend");
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async getByIdAsync(id) {
        try {
            const result = await this._httpClient.getAsync(`Friend/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async searchByUserId(id) {
        try {
            const result = await this._httpClient.getAsync(`Friend/searchByUserId/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async createAsync(data) {
        try {
            const result = await this._httpClient.postAsync("Friend", data);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }

    async deleteAsync(id) {
        try {
            const result = await this._httpClient.deleteAsync(`Friend/${id}`);
            return result;
        } catch (e) {
            console.log(e.message);
            return null;
        }
    }
}

export default FriendService;

File: src/CombatAnalysis.CutomerBL/Services/FriendService.cs
using AutoMapper;
using CombatAnalysis.CustomerBL.DTO;
using CombatAnalysis.CustomerBL.Interfaces;
using CombatAnalysis.CustomerDAL.Entities;
using CombatAnalysis.CustomerDAL.Interfaces;

namespace CombatAnalysis.CustomerBL.Services;

internal class FriendService : IService<FriendDto, int>
{
    private readonly IGenericRepository<Friend, int> _repository;
    private readonly IMapper _mapper;

    public FriendService(IGenericRepository<Friend, int> repository, IMapper mapper)
    {
        _repository = repository;
        _mapper = mapper;
    }

    public Task<FriendDto> CreateAsync(FriendDto item)
    {
        if (item == null)
        {
            throw new ArgumentNullException(nameof(FriendDto), $"The {nameof(FriendDto)} can't be null");
        }

        return CreateInternalAsync(item);
    }

    public async Task<int> DeleteAsync(int id)
    {
        var rowsAffected = await _repository.DeleteAsync(id);

        return rowsAffected;
    }

    public async Task<IEnumerable<FriendDto>> GetAllAsync()
    {
        var allData = await _repository.GetAllAsync();
        var result = _mapper.Map<List<FriendDto>>(allData);

        return result;
    }

    public async Task<FriendDto> GetByIdAsync(int id)
    {
        var result = await _repository.GetByIdAsync(id);
        var resultMap = _mapper.Map<FriendDto>(result);

        return resultMap;
    }

    public async Task<IEnumerable<FriendDto>> GetByParamAsync(string paramName, object value)
    {
        var result = await Task.Run(() => _repository.GetByParam(paramName, value));
        var resultMap = _mapper.Map<IEnumerable<FriendDto>>(result);

        return resultMap;
    }

    public Task<int> UpdateAsync(FriendDto item)
    {
        if (item == null)
        {
            throw new ArgumentNullException(nameof(FriendDto), $"The {nameof(FriendDto)} can't be null");
        }

        return UpdateInternalAsync(item);
    }


    private async Task<FriendDto> CreateInternalAsync(FriendDto item)
    {
        var map = _mapper.Map<Friend>(item);
        var createdItem = await _repository.CreateAsync(map);
        var resultMap = _mapper.Map<FriendDto>(createdItem);

        return resultMap;
    }

    private async Task<int> UpdateInternalAsync(FriendDto item)
    {
        var map = _mapper.Map<Friend>(item);
        var rowsAffected = await _repository.UpdateAsync(map);

        return rowsAffected;
    }
}
